### PR TITLE
[REFACTOR] Improve Application Header

### PR DESCRIPTION
**Pull request by olhasashchuk**
## Description  

This pull request refactors the `ApplicationHeader` component and introduces a new `MenuButton` for improved functionality and user experience. The changes enhance the code's maintainability, responsiveness, and alignment with UI design goals.  

- **Problem Solved:** Simplified management of application-level actions and improved the visual hierarchy of the application header.  
- **Impact:** Streamlined codebase, better UI responsiveness, and improved user interaction with key components.  

---  

### **Frontend:**  
1. **`ApplicationHeader.js`:**  
   - Added a new `MenuButton` component for contextual menu actions.  
   - Enhanced styling for better responsiveness and visual alignment.  
   - Updated to display application details like title, company name, status, and salary more effectively.  

2. **`MenuButton.js`:**  
   - Introduced a new component for managing actions like "Edit" and "Delete."  
   - Includes modals for editing (`ApplicationEditForm`) and deletion (`ApplicationDeleteModal`).  

3. **`CompanyEditForm.js`:**  
   - Added a note to inform users about the global impact of changes to company details.  
   - Updated styling for better user experience.  

4. **Deleted `ControlButton.js`:**  
   - Replaced its functionality with `MenuButton`.  

5. **Other File Updates:**  
   - Adjusted `page.js` to integrate `MenuButton` and simplify the layout.  
   - Minor style updates in `ModalWrapper.js`.  

---  

## Screenshots  

# 
Screenshot_12
Screenshot_11
Screenshot_13

---  

## Testing Steps  

1. **Frontend:**  
   - Verify that the `MenuButton` opens a menu with "Edit" and "Delete" options.  
   - Check that clicking "Edit" opens the `ApplicationEditForm` modal.  
   - Ensure "Delete" triggers the `ApplicationDeleteModal` with the appropriate messaging.  

2. **UI Interaction:**  
   - Confirm that the `ApplicationHeader` displays all details correctly and adapts to different screen sizes.  
   - Test the responsiveness and alignment of the new components across devices.

### CODE REVIEW

File: frontend/app/applications/[id]/ApplicationHeader.js [hunk 1]
```diff
@@ -111,28 +118,81 @@ export default function ApplicationHeader() {
...
+              fontSize: { xs: '1rem', sm: '1.25rem' },
+              fontWeight: 600,
+              textTransform: 'uppercase',
+            }}
+          >
+            {application.status}
+          </Typography>
+        </Stack>
+
+        {/* Salary */}
```
Comment (pankaj28843) [inline frontend/app/applications/[id]/ApplicationHeader.js:160]: You should remove this comment

**Review by pankaj28843 — APPROVED** [review]
Looks good to me, except that one small commented text.

Please ask your teammates to review the new UX/design.

**Comment by olhasashchuk** [comment]
Thank you! I had asked my teammates before created pull request.

**Review by pankaj28843 — APPROVED** [review]

**Review by Natata08 — APPROVED** [review]
Looks perfect!

**Pull request merged**